xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
I admit that in this code, for 1 break statement, it doesn't change much for readability. But when you have more complex things like loops within loops, lots of code being executed within a loop, etc. then `break` in the middle becomes a problem, and code written like the original code becomes a lot cleaner to read. Especially when there's lots of these conditions.

And according to the prof that I had back then, it also increases the chance of somehow making it even more sloppy. Will spare you the details ;-)